docs/source/user_guide/getting_started.rst

```diff
@@ -564,8 +564,74 @@ the programmer should not have to special case whether or not the value is
 present in the ``snapshot``. The ``snapshot`` is always built based on the schema
 and hence ``suite.snapshot.owner.credit`` would indeed be an attribute
 independently of whether the user has configured it. In this scenario the value
-of ``suite.snapshot.owner.credit`` would be ``None``. To configure the default
-value to something else then ``None``, we refer the reader to the next section.
+of ``suite.snapshot.owner.credit`` would be ``None``.
+
+Two issues arise from the behaviour of ``ConfigSuite``; what if a valid value is
+``None``? And how can default values be configured to something else then
+``None``? We try to facilitate both behaviours and address some solutions in
+the next sections.
+
+Allow None
+~~~~~~~~~~
+
+For certain configurations it may be reasonable for the user to provide
+``None`` as the value. The ``None`` is a specific value in the eyes of
+``ConfigSuite`` as it is also the value when a value hasn't been provided.
+As such, the schema must be specifically declared to allow the user to set
+``None`` in order for a configuration to be valid if it contains ``None``.
+
+The ``AllowNone`` type is only valid for ``BasicType``'s, not for containers.
+Setting ``AllowNone`` to ``True`` for anything but ``BasicType`` will result
+in an invalid schema.
+
+Let us see how the owner section of the ``cars`` schema could be configured
+in order for a configuration with ``None`` to pass.
+
+.. testcode:: [allownone]
+
+    import configsuite
+    from configsuite import types
+    from configsuite import MetaKeys as MK
+
+    schema = {
+        MK.Type: types.NamedDict,
+        MK.Content: {
+            "owner": {
+                MK.Type: types.NamedDict,
+                MK.Content: {
+                    "name": {MK.Type: types.String},
+                    "credit": {
+                        MK.Type: types.Number,
+                        MK.AllowNone: True,
+                    },
+                    "insured": {MK.Type: types.Bool},
+                },
+            },
+        },
+    }
+
+    config = {
+        "owner": {
+                "name": "Scrooge",
+                "credit": None,
+                "insured": False,
+        },
+    }
+
+    suite = configsuite.ConfigSuite(config, schema)
+    assert suite.valid
+
+    owner = suite.snapshot.owner
+    print("{} has a credit of {}".format(owner.name, owner.credit))
+
+.. testoutput:: [allownone]
+
+    Scrooge has a credit of None
+
+An important implication of allowing ``None`` is that it is impossible to know
+whether the value in the configuration was set to ``None`` explicitly, or if is
+just a result of a missing configuration item. As such ``ConfigSuite`` requires
```

Comment:
See the table in #70.

Reply:
I mean, the:
>ConfigSuite requires that any entity that is not required must also allow None

Is correct, is the reasoning also? I'm pretty sure it is, just thought it would be good with another look at it to make sure.

Reply:
I'm also pretty sure the:
>An important implication of allowing None is that it is impossible to know
whether the value in the configuration was set to None explicitly, or if is
just a result of a missing configuration item

is correct, but was this the only reason? The main reason?